### ISSUE TITLE
Fix #28

### ISSUE THREAD

**Issue by creazy231** [mancku/strapi-plugin-schemas-to-ts#30]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `strapi-plugin-schemas-to-ts@1.1.10` for the project I'm working on.

Fix #28 but introduces a **breaking change**

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/strapi-plugin-schemas-to-ts/dist/server/schemas-to-ts/interfaceBuilder.js b/node_modules/strapi-plugin-schemas-to-ts/dist/server/schemas-to-ts/interfaceBuilder.js
index 8a50b22..4cc41f1 100644
--- a/node_modules/strapi-plugin-schemas-to-ts/dist/server/schemas-to-ts/interfaceBuilder.js
+++ b/node_modules/strapi-plugin-schemas-to-ts/dist/server/schemas-to-ts/interfaceBuilder.js
@@ -130,14 +130,14 @@ class InterfaceBuilder {
       disconnect: T[];
     }
     `);
-        this.addCommonSchema(result, commonFolderModelsPath, 'BeforeRunEvent', `import { Event } from '@strapi/database/lib/lifecycles/index';
+        this.addCommonSchema(result, commonFolderModelsPath, 'BeforeRunEvent', `import { Event } from '@strapi/database/dist/lifecycles';
   
-    export interface BeforeRunEvent<TState> extends Event {
+    export interface BeforeRunEvent<TState extends Record<string, unknown>> extends Event {
       state: TState;
     }`);
         this.addCommonSchema(result, commonFolderModelsPath, 'AfterRunEvent', `import { BeforeRunEvent } from './BeforeRunEvent';
   
-    export interface AfterRunEvent<TState, TResult> extends BeforeRunEvent<TState> {
+    export interface AfterRunEvent<TState, TResult> extends BeforeRunEvent<TState extends Record<string, unknown> ? TState : never> {
       result: TResult;
     }
     `);
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by 20x-dz** [comment]
Thanks a lot @creazy231, the patch seems to work.

Any specific reason why you closed #29 though?

**Comment by creazy231** [comment]
@20x-dz I got this issue updating an older private project and had to make some quick changes. Just had no time to look deeper into this but after some other side effects I had to use the latest version of Strapi and therefore I implemented the initial change which caused other errors. Took these errors into AI which then generated me the changes you can see inside this patch. So overall fix was faster than initially expected.

**Comment by 20x-dz** [comment]
Ah, I didn't check the files of the PR, sorry about that. 🙈 

Would you mind reopening it or opening a new one with the patch you provided above and see if it gets merged? 🤞

**Comment by creazy231** [comment]
Sure, have to do some work but will open a new working PR later today.
But keep in mind that this is a breaking change and should be released in a new mayor version Y.X.X since most tools auto-update on X

**Comment by creazy231** [comment]
@20x-dz PR updated 🎉
https://github.com/mancku/strapi-plugin-schemas-to-ts/pull/29

**Comment by mancku** [comment]
Hi @20x-dz @creazy231 ! The PR #29 was merged and the new version of the plugin (1.1.20 is now in npm including this fix) Thanks a lot @creazy231 for your work in the PR!
